app/src/behaviors/behavior_input_two_axis.c

```diff
@@ -0,0 +1,270 @@
+    if (start == 0) {
+        return 0;
+    }
+    int64_t move_duration = now - (start + delay);
+    // start can be in the future if there's a delay
+    if (move_duration < 0) {
+        move_duration = 0;
+    }
+    return move_duration;
+}
+
+static float speed(const struct behavior_input_two_axis_config *config, float max_speed,
+                   int64_t duration_ms) {
+    // Calculate the speed based on MouseKeysAccel
+    // See https://en.wikipedia.org/wiki/Mouse_keys
+    if (duration_ms == 0) {
+        return 0;
+    }
+
+    if (duration_ms > config->time_to_max_speed_ms || config->time_to_max_speed_ms == 0 ||
+        config->acceleration_exponent == 0) {
+        return max_speed;
+    }
+    float time_fraction = (float)duration_ms / config->time_to_max_speed_ms;
+    return max_speed * powf(time_fraction, config->acceleration_exponent);
+}
+
+static void track_remainder(float *move, float *remainder) {
+    float new_move = *move + *remainder;
+    *remainder = new_move - (int)new_move;
+    *move = (int)new_move;
+}
+
+static float update_movement_1d(const struct behavior_input_two_axis_config *config,
+                                struct movement_state_1d *state, int64_t now) {
+    float move = 0;
+    if (state->speed == 0) {
+        state->remainder = 0;
+        return move;
+    }
+
+    int64_t move_duration = ms_since_start(state->start_time, now, config->delay_ms);
+    move =
+        (move_duration > 0)
+            ? (speed(config, state->speed, move_duration) * CONFIG_ZMK_MOUSE_TICK_DURATION / 1000)
+            : 0;
+
+    track_remainder(&(move), &(state->remainder));
+
+    return move;
+}
+static struct vector2d update_movement_2d(const struct behavior_input_two_axis_config *config,
+                                          struct movement_state_2d *state, int64_t now) {
+    struct vector2d move = {0};
+
+    move = (struct vector2d){
+        .x = update_movement_1d(config, &state->x, now),
+        .y = update_movement_1d(config, &state->y, now),
+    };
+
+    return move;
+}
+
+static bool is_non_zero_1d_movement(int16_t speed) { return speed != 0; }
+
+static bool is_non_zero_2d_movement(struct movement_state_2d *state) {
+    return is_non_zero_1d_movement(state->x.speed) || is_non_zero_1d_movement(state->y.speed);
+}
+
+static bool should_be_working(struct behavior_input_two_axis_data *data) {
+    return is_non_zero_2d_movement(&data->state);
+}
+
+static void tick_work_cb(struct k_work *work) {
+    struct k_work_delayable *d_work = k_work_delayable_from_work(work);
+    struct behavior_input_two_axis_data *data =
+        CONTAINER_OF(d_work, struct behavior_input_two_axis_data, tick_work);
+    const struct device *dev = data->dev;
+    const struct behavior_input_two_axis_config *cfg = dev->config;
+
+    uint32_t timestamp = k_uptime_get();
+
+    LOG_INF("tick start times: %lld %lld %lld", data->state.x.start_time, data->state.y.start_time,
+            timestamp);
```

Comment:
Build gives this warning here:
```
zmk/app/src/behaviors/behavior_input_two_axis.c:150:13: warning: format '%lld' expects argument of type 'long long int', but argument 4 has type 'uint32_t' {aka 'unsigned int'} [-Wformat=]
  150 |     LOG_INF("tick start times: %lld %lld %lld", data->state.x.start_time, data->state.y.start_time,
```

Reply:
In general, the logging isn't very clear unless you have the code in front of you. Maybe add some info? I.e. "tick times: x start XXX, y start XXX, current XXX".

Reply:
Indeed. Cleaned up and fixed the build warning.